include/seqan3/alphabet/structure/dot_bracket3.hpp

```diff
@@ -0,0 +1,326 @@
+
+    constexpr bool is_unpaired() const noexcept
+    {
+        return _value == internal_type::UNPAIRED;
+    }
+
+    //!\brief The ability of the alphabet to represent pseudoknots, i.e. crossing interactions.
+    static constexpr bool pseudoknot_support{false};
+    //!\}
+
+protected:
+    //!\privatesection
+    /*!\brief The internal type is a strictly typed enum.
+     *
+     * This is done to prevent aggregate initialization from numbers and/or chars.
+     * It is has the drawback that it also introduces a scope which in turn makes
+     * the static "letter values " members necessary.
+     */
+    enum struct internal_type : rank_type
+    {
+        UNPAIRED,
+        PAIR_OPEN,
+        PAIR_CLOSE,
+        UNKNOWN = UNPAIRED
+    };
+
+    //!\brief Value-to-char conversion table.
+    static constexpr char_type value_to_char[value_size]
+    {
+        '.',
+        '(',
+        ')'
+    };
+
+    //!\brief Char-to-value conversion table.
+    static constexpr std::array<internal_type, 256> char_to_value
+    {
+        [] () constexpr
+        {
+            std::array<internal_type, 256> rank_table{};
+
+            // initialize with UNKNOWN (std::array::fill unfortunately not constexpr)
+            for (internal_type & rnk : rank_table)
+                rnk = internal_type::UNKNOWN;
+
+            // canonical
+            rank_table['.'] = internal_type::UNPAIRED;
+            rank_table['('] = internal_type::PAIR_OPEN;
+            rank_table[')'] = internal_type::PAIR_CLOSE;
+
+            return rank_table;
+        } ()
+    };
+
+public:
+    //!\privatesection
+    //!\brief The data member.
+    internal_type _value;
+    //!\publicsection
+};
+
+constexpr dot_bracket3 dot_bracket3::UNPAIRED{internal_type::UNPAIRED};
+constexpr dot_bracket3 dot_bracket3::PAIR_OPEN{internal_type::PAIR_OPEN};
+constexpr dot_bracket3 dot_bracket3::PAIR_CLOSE{internal_type::PAIR_CLOSE};
+constexpr dot_bracket3 dot_bracket3::UNKNOWN{internal_type::UNKNOWN};
+
+} // namespace seqan3
+
+#ifndef NDEBUG
+static_assert(seqan3::alphabet_concept<seqan3::dot_bracket3>);
+static_assert(seqan3::rna_structure_concept<seqan3::dot_bracket3>);
+#endif
+
+// ------------------------------------------------------------------
+// literals
+// ------------------------------------------------------------------
+
+namespace seqan3::literal
+{
+
+/*!\brief dot_bracket3 literal
+ * \relates seqan3::dot_bracket3
+ * \returns std::vector<seqan3::dot_bracket3>
+ *
+ * You can use this string literal to easily assign to a vector of dot_bracket3 characters:
+ *
+ *```.cpp
+ *     using namespace seqan3::literal;
+ *     std::vector<dot_bracket3> foo{".(..)."_dot_bracket3};
+ *     std::vector<dot_bracket3> bar = ".(..)."_dot_bracket3;
+ *     auto bax = ".(..)."_dot_bracket3;
+ *```
+ *
+ * \attention
+ * All seqan3 literals are in the namespace seqan3::literal!
+ */
+inline std::vector<dot_bracket3> operator "" _dot_bracket3(const char * str, std::size_t len)
```

Comment:
maybe abbreviate this to `_db3` ? If it is too long there really is no point in using it ;-)

Reply:
Yes, it is indeed too long. I change it to `_db3`.